### PR TITLE
terminating streams with onError when connection is closed

### PR DESCRIPTION
**Pull request by lehecka**

### CODE REVIEW

File: src/Common.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "src/Common.h"
+#include <glog/logging.h>
+#include <folly/io/IOBuf.h>
+#include "src/AbstractStreamAutomaton.h"
+
+namespace reactivesocket {
+
+static const char* getTerminatingSignalErrorMessage(int terminatingSignal) {
```
Comment (jspahrsummers) [inline src/Common.cpp]: Why doesn't this accept the type `StreamCompletionSignal`?

File: src/Common.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  switch ((StreamCompletionSignal)terminatingSignal) {
+    case StreamCompletionSignal::CONNECTION_END:
+      return "connection closed";
+    case StreamCompletionSignal::CONNECTION_ERROR:
+      return "connection error";
+    case StreamCompletionSignal::ERROR:
+      return "general error";
+    case StreamCompletionSignal::GRACEFUL:
+      DCHECK(false) << "throwing exception for GRACEFUL termination?";
```
Comment (jspahrsummers) [inline src/Common.cpp]: Can this still return a string saying "graceful termination," just in case we need to debug it in production?

Comment (lehecka) [inline src/Common.cpp]: yes, it can

**Comment by yschimke** [comment]
Can we land this also before importing, this hit me

**Pull request merged**